lib/taskbook.js

```diff
@@ -333,6 +348,21 @@ class Taskbook {
     render.markIncomplete(unchecked);
   }
 
+  beginTasks(ids) {
+    ids = this._validateIDs(ids);
+    const {_data} = this;
+    const [started, stopped] = [[], []];
+
+    ids.forEach(id => {
+      _data[id].inProgress = !_data[id].inProgress;
+      return _data[id].inProgress ? started.push(id) : stopped.push(id);
+    });
```

Comment:
We should also avoid the case where both the `isComplete` and `inProgress` attributes are set to `true`, something that can be done by setting the `isComplete`  attribute of each iterated item to `false`, thus achieving mutual exclusion:

```js
    ids.forEach(id => {
     _data[id].isComplete = false; 
     _data[id].inProgress = !_data[id].inProgress;
      return _data[id].inProgress ? started.push(id) : paused.push(id);
    });
```

The same should be done for the `checkTasks(ids)` method, on line 342, where `inProgress` should be set to `false`.